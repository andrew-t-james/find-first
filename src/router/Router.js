import React from 'react';
import { BrowserRouter, Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import Home from '../Components/Home/Home';
import Dashboard from '../Components/Dashboard/Dashboard';
import SignIn from '../Components/SignIn/SignIn';
import JobDetail from '../Components/JobDetail/JobDetail';


export const AppRouter = ({ jobs }) => (
  <BrowserRouter>
    <Switch>
      <Route path="/" exact component={Home} />
      <Route path="/dashboard" exact component={Dashboard} />
      <Route path="/sign-in" exact component={SignIn} />
      <Route path="/job/:id" render={({ match }) => {
        const job = jobs.find(job => job.id === Number(match.params.id));
        return (
          <div>
            <JobDetail {...job}/>
          </div>
        );
      }} />
    </Switch>
  </BrowserRouter>
);

const mapStateToProps = state => ({
  jobs: state.githubJobs
});

export default connect(mapStateToProps)(AppRouter);
