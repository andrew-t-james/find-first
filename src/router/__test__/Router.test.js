import React from 'react';
import { shallow } from 'enzyme';
import Router from '../Router';

describe('<Router />', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Router />));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
