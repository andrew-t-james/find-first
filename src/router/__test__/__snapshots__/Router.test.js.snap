// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Router /> should match snapshot 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/dashboard"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/sign-in"
    />
  </Switch>
</BrowserRouter>
`;
